docs/tutorials/forecasting/hierarchical_model_tutorial.md.template

```diff
@@ -0,0 +1,177 @@
+# Hierarchical Model Tutorial
+
+
+This tutorial illustrates how to use GluonTS' deep-learning based hierarchical model
+`DeepVarHierarchical`. We first explain the data preparation for hierarchical/grouped time series,
+and then show the model training, prediction and evaluation using common use-cases.
+
+
+## Introduction
+
+The important aspect in using hierarchical models is the proper preparation of hierarchical
+time series data. The minimal requirements in building the hierarchical or grouped time series
+are the time series at the bottom-level of the hierarchy and the hierarchical/grouped
+aggregation matrix.
+`GluonTS` provides a simple way to construct hierarchical time series by reading bottom-level
+time series and the aggregation matrix from csv files.
+Here we first describe the preparation of hierarchical time series before discussing the
+training of the hierarchical model.
+
+
+## Preparation of Hierarchical Time Series
+
+The bottom level time series are assumed to be available in a csv file as columns.
+The csv file should also contain the index column listing the time stamps for each row.
+Note that the aggregated time series are automatically constructed and should not be provided.
+Here is an [example csv](https://gist.githubusercontent.com/rshyamsundar/39e57075743537c4100a716a7b7ec047/raw/f02f5aeadad73e3f3e9cf54478606d3507826492/example_bottom_ts.csv) of bottom level time series.
+Similarly, the aggregation matrix can also be read from a csv file;
+here is an [example](https://gist.githubusercontent.com/rshyamsundar/17084fd1f28021867bcf6f2d69d9b73a/raw/32780ca43f57a78f2d521a75e73b136b17f34a02/example_agg_mat.csv).
+Note that grouped time series can also be represented by an aggregation matrix in the same
+way as the hierarchical time series and hence the material presented here is also applicable to
+grouped time series.
+
+
+```python
+import pandas as pd
+from gluonts.dataset.hierarchical import HierarchicalTimeSeries
+
+# Load (only!) the time series at the bottom level of the hierarchy.
+ts_at_bottom_level_csv = (
+    "https://gist.githubusercontent.com/rshyamsundar/39e57075743537c4100a716a7b7ec047/"
+    "raw/f02f5aeadad73e3f3e9cf54478606d3507826492/example_bottom_ts.csv"
+)
+ts_at_bottom_level = pd.read_csv(
+    ts_at_bottom_level_csv,
+    index_col=0,
+    parse_dates=True,
+)
+
+# Make sure the dataframe has `PeriodIndex`.
+ts_at_bottom_level = ts_at_bottom_level.to_period()
```

Comment:
```suggestion
ts_at_bottom_level = pd.read_csv(
    ts_at_bottom_level_csv,
    index_col=0,
    parse_dates=True,
).to_period()
```

The fact that a `PeriodIndex` is needed could be stressed in the text

Reply:
I had your suggested version initially and thought people might miss `to_period` call if it is in a single statement and raise unnecessary issues around this point. So made that explicit. I will change and stress it in the text.

Reply:
Done